Makefile

```diff
@@ -113,6 +113,69 @@ distclean:
 
 check: test
 
+# Remove files generated by running coverage, put the non-instrumented lib back 
+# in place 
+coverage-clean:
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	-rm -rf gcovr
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp coverage
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+
+# Build and test with code coverage reporting.  Leave the lib directory 
+# instrumented for any additional runs the user may want to make.
+# For C++ coverage reporting, this needs to be run in conjunction with configure
+#  --coverage.  html coverage reports will be created under coverage/
+
+coverage: coverage-build coverage-test
+
+coverage-build: all
+	mkdir node_modules
+	$(NODE) ./deps/npm install istanbul-merge
```

Comment:
@addaleax, no just the idea that if its already downloaded, that you could continue to run make coverage without additional downloads being required.  So check if you have already downloaded what's necessary so that the 2nd run an on don't necessarily need that.

I also think that is something we can break out into a follow on PR so that people can start using